include/fmt/std.h

```diff
@@ -91,6 +94,43 @@ template <typename Char>
 struct formatter<std::thread::id, Char> : basic_ostream_formatter<Char> {};
 FMT_END_NAMESPACE
 
+#ifdef __cpp_lib_optional
+FMT_BEGIN_NAMESPACE
+template <typename T, typename Char>
+struct formatter<std::optional<T>, Char,
+                 std::enable_if_t<is_formattable<T>::value>> {
+  formatter<T, Char> underlying_;
+  static constexpr basic_string_view<Char> OPTION =
+      detail::string_literal<Char, 'o', 'p', 't', 'i', 'o', 'n', 'a', 'l', '('>{};
+  static constexpr basic_string_view<Char> EMPTY =
+      detail::string_literal<Char, 'n', 'o', 'n', 'e'>{};
```

Comment:
Let's make these names consistent with the content and fix naming conventions (we use lowercase for constants): OPTION -> optional, EMPTY -> none.